main.go

```diff
@@ -1398,6 +1425,385 @@ sent as DATA.
+			}
+
+			if opt.X509.Validate.Bits != nil {
+				if err = cert.CheckStrength(opt.X509.Validate.Bits...); err != nil {
+					return fmt.Errorf("%s is fails strength requirement: %s", path, err)
+				}
+			}
+
+			if opt.X509.Validate.CA && !cert.IsCA() {
+				return fmt.Errorf("%s is not a certificate authority", path)
+			}
+
+			if opt.X509.Validate.Revoked && !ca.HasRevoked(cert) {
+				return fmt.Errorf("%s has not been revoked by %s", path, opt.X509.Validate.SignedBy)
+			}
+			if opt.X509.Validate.NotRevoked && ca.HasRevoked(cert) {
+				return fmt.Errorf("%s has been revoked by %s", path, opt.X509.Validate.SignedBy)
+			}
+
+			if opt.X509.Validate.Expired && !cert.Expired() {
+				return fmt.Errorf("%s has not yet expired", path)
+			}
+			if opt.X509.Validate.NotExpired && cert.Expired() {
+				return fmt.Errorf("%s has expired", path)
+			}
+
+			if _, err = cert.ValidFor(opt.X509.Validate.Name...); err != nil {
+				return err
+			}
+
+			ansi.Printf("@G{%s} checks out.\n", path)
+		}
+
+		return nil
+	})
+
+	r.Dispatch("x509 issue", &Help{
+		Summary: "Issue X.509 Certificates and Certificate Authorities",
+		Usage:   "safe x509 issue [OPTIONS] --name cn.example.com path/to/certificate",
+		Description: `
+Issue a new X.509 Certificate
+
+The following options are recognized:
+
+  -A, --ca          This certificate is a CA, and can
+                    sign other certificates.
+
+  -s, --subject     The subject name for this certificate.
+                    i.e. /cn=www.example.com/c=us/st=ny...
+                    If not specified, the first '--name'
+                    will be used as a lone CN=...
+
+  -i, --signed-by   Path in the Vault where the CA certificate
+                    (and signing key) can be found.
+                    Without this option, 'x509 issue' creates
+                    self-signed certificates.
+
+  -n, --name        Subject Alternate Name(s) for this
+                    certificate.  These can be domain names,
+                    IP addresses or email address -- safe will
+                    figure out how to properly encode them.
+                    Can (and probably should) be specified
+                    more than once.
+
+  -b, --bits N      RSA key strength, in bits.  The only valid
+                    arguments are 1024 (highly discouraged),
+                    2048 and 4096.  Defaults to 4096.
+
+  -t, --ttl         How long the new certificate will be valid
+                    for.  Specified in units h (hours), m (months)
+                    d (days) or y (years).  1m = 30d and 1y = 365d
+                    Defaults to 10y
+	`,
+	}, func(command string, args ...string) error {
+		rc.Apply()
+
+		var ca *vault.X509
+
+		if len(args) != 1 || len(opt.X509.Issue.Name) == 0 {
+			r.ExitWithUsage("x509 issue")
+		}
+
+		if opt.X509.Issue.Subject == "" {
+			opt.X509.Issue.Subject = fmt.Sprintf("CN=%s", opt.X509.Issue.Name[0])
```

Comment:
is that actually invalid? seems like if you wanted a regular cert to be just valid for an IP address, and you connect via IP address, it should be allowed. If they try to connect via a name then, and the name is in the SANs, it should still be valid.

I guess.. is there anything in the RFC saying that the CN cannot be an IP addr?